openlibrary/plugins/worksearch/publishers.py

```diff
@@ -60,30 +60,26 @@ class publisher_search(delegate.page):
 
     def GET(self):
         i = web.input(q="")
-        solr = search.get_solr()
-        q = {"publisher": i.q}
-
-        result = solr.select(
-            q,
+        result = search.get_solr().select(
+            {"publisher": i.q, "type": "work"},
             facets=["publisher_facet"],
-            fields=["publisher", "publisher_facet"],
+            facet_mincount=1,
+            facet_limit=25,
```

Comment:
Make it filter to only facets that contain the query
```suggestion
            facet_limit=25,
            facet_contains=i.q,
            facet_contains_ignoreCase=True,
```